src/main/java/br/com/metricminer2/scm/GitRepository.java

```diff
@@ -410,4 +422,21 @@ public String blame(String file, String commitToBeBlamed, Integer line) {
 		}
 	}
 
+	public int getMaxNumberFilesInACommit() {
+		if(this.maxNumberFilesInACommit == null){
+			return DEFAULT_MAX_NUMBER_OF_FILES_IN_A_COMMIT;
+		} else {
+			return this.maxNumberFilesInACommit;
+		}
+	}
+
+	@Override
+	public void setMaxNumberFilesInACommit(int maxNumber) {
```

Comment:
I usually prefer to avoid setters in cases which 1) it's a configuration that never changes in the future, 2) it's kinda important since the beginning of the class.

What about receiving this "maxNumber" and the "masterBranch" in the constructor, and then use the existing static factory methods (+ new ones) to allow multiple options? 

Also, with these two new attributes, we can't configure the study using the fluent interface anymore, as this Git initialization would require a few lines.